### PR TITLE
Use atomic writer to cache files to fix race condition

### PR DESCRIPTION
**Pull request by gijskoning**
Race condition happened in process_replay. Might also have happened in Mesh3d!

Saving the compressed file is not necessary so its removed.
Decompresses in memory and saves it using atomic writer.

### CODE REVIEW

File: laika/downloader.py [hunk 1]
```diff
@@ -253,38 +255,37 @@ def download_and_cache_file_return_first_success(url_bases, folder_and_file_name
...
     try:
       data_zipped = download_file(url_base, folder_path, filename_zipped)
     except (IOError, pycurl.error, socket.timeout):
       unix_time = time.time()
       tmp_dir = cache_dir + '/tmp'
       os.makedirs(tmp_dir, exist_ok=True)
-      with tempfile.NamedTemporaryFile(delete=False, dir=tmp_dir) as fout:
-        fout.write(str.encode(str(unix_time)))
-      os.replace(fout.name, filepath + '.attempt_time')
+      with atomic_write(tmp_dir, mode='wb') as wf:
```
Comment (pd0wm) [inline laika/downloader.py:277]:
```suggestion
      with atomic_write(filepath_attempt, mode='wb') as wf:
```

Comment (pd0wm) [reply laika/downloader.py:277]: And the while `tmp_dir` is no longer necessary

File: laika/downloader.py [hunk 2]
```diff
@@ -376,16 +373,11 @@ def download_prediction_orbits_russia_src(gps_time, cache_dir):
   # Example: Stark_1D_22060100.sp3
   folder_and_file_names = [(folder_path, file_prefix + f"{h:02}.sp3") for h in reversed(current_day)] + \
                           [(folder_path_prev, file_prefix_prev + f"{h:02}.sp3") for h in reversed(prev_day)]
-  file = download_and_cache_file_return_first_success(url_bases, folder_and_file_names, cache_subdir)
-  if file is None: # todo remove this when gnss-alt repo is fixed
```
Comment (pd0wm) [inline laika/downloader.py:380]: Is this fixed? Maybe something for a separate PR in case this needs to be reverted.

File: laika/downloader.py [hunk 3]
```diff
@@ -253,38 +255,35 @@ def download_and_cache_file_return_first_success(url_bases, folder_and_file_name
...
       data_zipped = download_file(url_base, folder_path, filename_zipped)
     except (IOError, pycurl.error, socket.timeout):
       unix_time = time.time()
-      tmp_dir = cache_dir + '/tmp'
-      os.makedirs(tmp_dir, exist_ok=True)
-      with tempfile.NamedTemporaryFile(delete=False, dir=tmp_dir) as fout:
-        fout.write(str.encode(str(unix_time)))
-      os.replace(fout.name, filepath + '.attempt_time')
+      with atomic_write(filepath_attempt, mode='wb') as wf:
+        wf.write(str.encode(str(unix_time)))
```
Comment (pd0wm) [inline laika/downloader.py:276]:
```suggestion
      with atomic_write(filepath_attempt, mode='w') as wf:
        wf.write(str(unix_time))
```

Comment (gijskoning) [reply laika/downloader.py]: Thanks! I also cleaned up reading the attempt file

**Pull request merged**